### PR TITLE
Retro active BP entry (v1)

### PR DESCRIPTION
**Pull request by saket**
https://www.pivotaltracker.com/story/show/163052502

Large-ish PR because the new (and incomplete) date entry functionality is behind a feature flag and it involved duplicating a few classes.

### CODE REVIEW

**Comment by saket** [comment]
@sanchita please skip the last commit, `Validate date and save BP when entering a new BP`. I'm moving that to a future branch.

File: app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetEvents.kt [hunk 1]
```diff
@@ -21,3 +21,10 @@ object BloodPressureRemoveClicked : UiEvent {
 }
 
 object BloodPressureDiastolicBackspaceClicked : UiEvent
+
+data class BloodPressureScreenChanged(val type: ScreenType) : UiEvent
+
+enum class ScreenType {
+  BP_ENTRY,
+  DATE_ENTRY
```
Comment (sanchita) [inline app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetEvents.kt]: I am guessing this `enum` will be used elsewhere.
Can we extract it to its own file?

File: app/src/debug/java/org/simple/clinic/DebugClinicApp.kt [hunk 2]
```diff
@@ -112,9 +112,9 @@ class DebugClinicApp : ClinicApp() {
           }
```
[the remark this answers is not in the document]
Comment (rakshakhegde) [reply app/src/debug/java/org/simple/clinic/DebugClinicApp.kt]: Sure

File: app/src/main/java/org/simple/clinic/util/KotlinFoo.kt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 package org.simple.clinic.util
 
 // Forces when blocks to be exhaustive.
-fun Unit.exhaustive() {}
+fun Any?.exhaustive() {}
```
[the remark this answers is not in the document]
Comment (rakshakhegde) [reply app/src/main/java/org/simple/clinic/util/KotlinFoo.kt]: Right, we don't need this change anymore. I've deleted this commit.

File: app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetEvents.kt (hunk 1 above)
Comment (rakshakhegde) [reply app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetEvents.kt]: I don't think we'll need to use anywhere else. Let's keep this here until we use it elsewhere? What do you think?

Comment (saket) [reply app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetEvents.kt]: Yep, just here

File: app/src/main/res/layout/sheet_blood_pressure_entry.xml [hunk 4]
```diff
@@ -64,9 +65,10 @@
     <EditText
       android:id="@+id/bloodpressureentry_systolic"
       style="@style/Clinic.V2.BloodPressureInput"
-      android:layout_toStartOf="@+id/bloodpressureentry_systolic_diastolic_separator"
+      android:layout_marginEnd="@dimen/spacing_12"
```
[the remark this answers is not in the document]
Comment (saket) [reply app/src/main/res/layout/sheet_blood_pressure_entry.xml]: Lint checks are always opt-out. We manually disable a few of them in `lint.xml`, but everything else should be enabled by default. 

Are you seeing `MissingConstraints` lint warning on your side?

File: app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetEvents.kt (hunk 1 above)
Comment (saket) [reply app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheetEvents.kt]: Update: I was wrong, this needs to be moved out of `BloodPressureEntrySheetEvents`. I'll move it to the screen.

File: build.gradle [hunk 5]
```diff
@@ -52,6 +52,7 @@ buildscript {
       javaStringSimilarity: '1.1.0',
       espresso            : '3.1.1',
       leakCanary          : '1.6.2',
+      constraintLayout    : '1.1.2',
```
Comment (rakshakhegde) [inline build.gradle]: The latest version of `androidx.constraintlayout` is `1.1.3`. Let's bump this up?

Comment (saket) [reply build.gradle]: We'll do this later. Let's favor merging branches asap and adding updates to backlog.

Comment (saket) [reply build.gradle]: Considering that this is PR for an ongoing, changes like these can go in upcoming PRs

**Pull request merged**